kuzzleio/kuzzle
# Add events on http routing

## What does this PR do ?


This PR add 6 events related to the HTTP method used. `http:get`, `http:post`, `http:put`, `http:patch`, `http:delete`, `http:head` (`http:options` was already present).  
See [Jira#KZL-42](https://jira.kaliop.net/browse/KZL-42) or https://github.com/kuzzleio/kuzzle/issues/1099 for more context.  

The events are fired in `Router.route()` and now this method always return a promise.  

Changes are documented here : https://github.com/kuzzleio/documentation/pull/478

### How should this be manually tested?

  - Step 1 : Run a Kuzzle stack with plugins debug : `DEBUG=kuzzle:plugins docker-compose -f docker-compose/dev.yml up`
  - Step 2 : Try running a curl request and verify if the correct event is fired : `curl -X PUT localhost:7512`
![image](https://user-images.githubusercontent.com/4447392/39704970-480dd522-520d-11e8-9e4c-ba5019549e15.png)

  - Step 3 : Repeat with all HTTP methods: `methods=(GET POST PUT PATCH DELETE HEAD); for method in $methods; do curl -X $method localhost:7512/_getInfo; done`

### Boyscout

 - Fix listener arguments on `unhandledRejection` (Related to #1020 )
 - Fix `should().be.called()` syntax in tests
 - Allow user to remove `node_modules` created by docker without `sudo`
 - Replace references to back-office by admin console
